whole-tale/terraform_deployment
# Preliminary backup implementation using rclone

Along with repo https://github.com/whole-tale/backup This PR fixes  https://github.com/whole-tale/terraform_deployment/issues/11

Changes:
* Adds init-backup.sh script to create systemd service and timer for daily backups on fileserver
* Removes restore_url capability.  Restore now requires manual execution from within the backup container
* Changes wt_mongo network to be attachable

Test case:
* Precondition: Assumes new deployment (I've been testing on Nebula in NDS project)
* Configure rclone, following instructions in README.md to configure for Box. (This is somewhat cumbersome, but so far the easiest way):
```
cd terraform_deployment
rclone --config rclone.conf config
```
 * Configure `variables.tf` as per usual, then run `terraform apply`
* `ssh <fileserver>` and  confirm `/etc/systemd/system/backup.service` and `backup.timer` are present
* Restore from a previous dev backup then use the `init-mongo.sh` to replace the domain and globus credentials. This step requires restarting Girder.
```
$ docker run --rm --network wt_mongo -v /mnt/home:/backup -v /home/core/rclone/:/conf wholetale/backup bash
# restore.sh -c dev -d 20180416
# init-mongo.sh <domain> <globus_client_id> <globus_client_secret>
```
* Confirm restore worked by logging into your instance
* Run a backup.  Either wait for cron task to execute or manually run command `docker run --rm --network wt_mongo -v /mnt/home:/backup -v /home/core/rclone/:/conf wholetale/backup backup.sh -c cluster_id` where `cluster_id` is a unique name for your cluster.  This will create in Box WT/<cluster_id>/<date> containing tgz files for both home directories and mongodb.
* Confirm files are present via rclone (in backup container)
```
rclone --config /conf/rclone.conf ls backup:WT/<cluster_id>/<date>
```

By default, it will only retain backups for 7 days.
